### DISCUSSION TITLE
Is it possible to get the next buildId as an env?

### DISCUSSION THREAD

**Discussion by AssisrMatheus** [vercel/next.js#14743] [Help]
I'm logging with sentry and following the [with-sentry](https://github.com/vercel/next.js/tree/canary/examples/with-sentry) example.

But I want to add the `release` property to my `Sentry.init` call. The example uploads the sourcemaps using the `option.buildId` as its `release` propery in the [next.config.js](https://github.com/vercel/next.js/blob/cbb1c5c75552d7ce29846073f3103f050e9da641/examples/with-sentry/next.config.js#L54) file. And I want to match that when running `Sentry.init`.

Is it possible to have the buildId as an env variable in both server and client?

**Comment by AssisrMatheus** [comment]
I currently added:
```
webpack: (config, { webpack, buildId, isServer }) => {
    config.plugins.push(
      new webpack.DefinePlugin({
        'process.env.CONFIG_BUILD_ID': JSON.stringify(buildId)
      })
    );
}
```

To my `next.config.js` and then I can read it with `process.env.CONFIG_BUILD_ID`. But is that a good practice?

**Reply by AubreyHewes** [reply]
@AssisrMatheus you will need to return the `config` in example else the dev/build will fail...

I do exactly the same as you (though returning the config :wink:) to get this information for displaying the build within an app..

If it is good practice?

> custom webpack configuration is not guaranteed https://github.com/vercel/next.js/issues/12944#issuecomment-629407277


I agree it is useful to be available i.e. just for displaying the current `buildId` in an app. Adding to the env a possible PR?

For sentry I tend to think it is not actually required as after you register a release (what the plugin does?) all events are on the latest release..? Though I could be wrong; if you have multiple workers and one has not been updated then it would incorrectly log to the latest release?... so would require it's build release _(just musing on sentry integration)_

**Comment by JonasDoesThings** [comment]
Is there any update for that? I've checked the process.env fields, but to no avail. 
So is the build id still not available to use in the frontend?
